mobile/src/main/java/org/openhab/habdroid/ui/preference/DeviceIdentifierPreference.kt

```diff
@@ -0,0 +1,166 @@
+            this.value = value
+            updateSummary()
+        }
+    }
+
+    fun createDialog(): DialogFragment {
+        return PrefFragment.newInstance(key, title)
+    }
+
+    class PrefFragment : PreferenceDialogFragmentCompat(), TextWatcher {
+        private lateinit var editorWrapper: TextInputLayout
+        private lateinit var editor: EditText
+        private lateinit var voiceButton: SwitchMaterial
+        private lateinit var backgroundTasksButton: SwitchMaterial
+
+        override fun onCreateDialogView(context: Context?): View {
+            val inflater = LayoutInflater.from(activity)
+            val v = inflater.inflate(R.layout.pref_dialog_device_identifier, null)
+
+            editorWrapper = v.findViewById(R.id.input_wrapper)
+            editor = v.findViewById(android.R.id.edit)
+            voiceButton = v.findViewById(R.id.voice_switch)
+            backgroundTasksButton = v.findViewById(R.id.background_tasks_switch)
+
+            editor.addTextChangedListener(this)
+            editor.inputType = InputType.TYPE_TEXT_FLAG_NO_SUGGESTIONS
+            arguments?.getCharSequence(KEY_TITLE)?.let { title ->
+                editorWrapper.hint = title
+            }
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                editor.importantForAutofill = View.IMPORTANT_FOR_AUTOFILL_NO
+            }
+
+            val prefs = preference.sharedPreferences
+            editor.setText(((preference as DeviceIdentifierPreference).value as String?))
+            editor.setSelection(editor.text.length)
+            voiceButton.isChecked = prefs.getBoolean(PrefKeys.DEV_ID_PREFIX_VOICE, false)
+            backgroundTasksButton.isChecked = prefs.getBoolean(PrefKeys.DEV_ID_PREFIX_BG_TASKS, true)
+
+            return v
+        }
+
+        override fun onDialogClosed(positiveResult: Boolean) {
+            if (positiveResult) {
+                val prefs = preference.sharedPreferences
+                prefs.edit {
+                    val pref = preference as DeviceIdentifierPreference
+                    pref.setValue(editor.text.toString())
+                    putBoolean(PrefKeys.DEV_ID_PREFIX_VOICE, voiceButton.isChecked)
+                    putBoolean(PrefKeys.DEV_ID_PREFIX_BG_TASKS, backgroundTasksButton.isChecked)
+                }
+            }
+        }
+
+        override fun onStart() {
+            super.onStart()
+            updateOkButtonState()
+        }
+
+        override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {
+            // no-op
+        }
+
+        override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {
+            // no-op
+        }
+
+        override fun afterTextChanged(s: Editable) {
+            val value = s.toString()
+            if (value.contains(" ") || value.contains("\n")) {
+                editorWrapper.error = context?.getString(R.string.error_sending_alarm_clock_item_empty)
```

Comment:
That used string resource looks a little weird here ... what does alarm clock have to do with this?

Reply:
The string res resolves to `Please enter a valid Item name`. I can rename it, though.